.gitignore

```diff
@@ -11,3 +11,4 @@ apidoc/
 .ipynb_checkpoints/
 test_times.dat
 core.*
+.vscode/settings.json
```

Comment:
```suggestion
```

Or does this have a particular reason?

Reply:
I use vscode and it usually pushes changes in my settings to git sometimes

Reply:
I think it is very helpful for all vscode users, just like including the `.idea/` folder in the `.gitignore` is helpful for Pycharm users. So we could debate if this should be a separate pull request, but in general I think it is a helpful addition.